backupd/src/storage/mod.rs

```diff
@@ -16,8 +20,9 @@ impl FileLen for File {
 
 pub trait StorageManager<'a> {
     type File: Read + Write + Seek + FileLen + 'a;
-    fn create_storage(&'a self, path: String) -> Result<Self::File, String>;
+    fn create_storage(&'a self, metadata: &FileMetadata) -> Result<Self::File, String>;
```

Comment:
The function and argument names could probably be better - I can't even tell what they're supposed to do based on this.

Is it supposed to be: save, find, exists?

Also it doesn't seem like a great idea that these functions return file pointers? How about e.g.

```
create(self, metadata, chunk) -> Open a file and insert the first bytes to it, save length etc. to DB
append(self, metadata, index, chunk) -> Open a file, and write the given chunk starting from the `index` assuming `index <= file_length`
read(self, metadata, index, num_bytes) -> Read a chunk of a file
exists(self, metadata) -> Return boolean for a match + number of bytes written
delete(self, metadata) -> Delete if exists
```

Reply:
It's creation, get, and is_outdated. I can discuss more about fixing the naming. It doesn't need append or read as the structure returned implements read, write, and seek.

It doesn't return a File pointer, it returns an object/structure that implements Read, Write, Seek, and FileLen. It has a lifetime bound in case it borrows any data from the StorageManager itself (as is the case with in-memory StorageManager).